src/Core/src/Eventuous.Subscriptions/Logging/InternalLogger.cs

```diff
@@ -2,16 +2,27 @@
 // Licensed under the Apache License, Version 2.0.
 
 using Microsoft.Extensions.Logging;
+using System.Collections.Immutable;
 
 // ReSharper disable TemplateIsNotCompileTimeConstantProblem
 
 namespace Eventuous.Subscriptions.Logging;
 
 public class InternalLogger(ILogger logger, LogLevel logLevel, string subscriptionId) {
+    readonly ImmutableDictionary<string, object> _scope = ImmutableDictionary<string, object>.Empty.Add("SubscriptionId", subscriptionId);
+
 #pragma warning disable CA2254
-    public void Log(string message, params object[] args) => logger.Log(logLevel, GetMessage(message), args);
+    public void Log(string message, params object[] args) {
+        using (logger.BeginScope(_scope)) {
+            logger.Log(logLevel, GetMessage(message), args);
+        }
+    }
 
-    public void Log(Exception? exception, string message, params object[] args) => logger.Log(logLevel, exception, GetMessage(message), args);
+    public void Log(Exception? exception, string message, params object[] args) {
+        using (logger.BeginScope(_scope)) {
```

Comment:
Isn't it adding quite a few allocations? What the purpose of the scope here, I don't fully understand. I would understand a scope for handling each message, which starts in the subscription, and ends in the consumer. But, creating and disposing the scope for logging one log message? Is it for adding the subscription id property to the log message rather than just having it as a string? It's totally possible to do at the subscription level as it creates the log context when it starts. I think scopes are explained in the docs as "a logical operation", which is not the case here.

Reply:
You're right - I hadn't thought about putting it earlier in the flow. I'll work on doing that instead of this

Reply:
I have now moved the log scope into the EventSubscription and the DefaultConsumer - as far as I see it that should minimize the allocations and be more in line with the logical operation aspect